### DISCUSSION TITLE
Problem with interrupt wdt causing problems when in sleep mode

### DISCUSSION THREAD

**Discussion by XanCraft21** [espressif/arduino-esp32#8206] [Q&A]
Hello, i have been having problems with the sleep functions, and it affects both light and deep sleep modes.

The interrupt watchdog timer keeps crashing the esp every time after entering sleep mode with repeated software resets until a power on reset happens.

There seems to be absolutely no info online about how to disable the interrupt watchdog for Arduino and it’s important that i find a way how to disable the timer. I’ve tried many different things and still no luck.

If any of the workers developers or contributors know how i can disable this please let me know. I need to know how to disable it because i am working on a major project that requires at least light sleep mode for battery power saving.

**Comment by mrengineer7777** [comment]
Can you run this example without crashing? https://github.com/espressif/arduino-esp32/blob/master/libraries/ESP32/examples/DeepSleep/TimerWakeUp/TimerWakeUp.ino

**Reply by XanCraft21** [reply]
Yes. It appears to be working fine with a timer. But my problem comes from using a gpio interrupt wake up. It begins to glitch when using the interrupt mode and gets stuck that way. I think i even did it without the sleep begin command but it still bugged after setting the interrupt mode.

**Reply by mrengineer7777** [reply]
Ah so the problem may have nothing to do with Sleep.  Try creating a simple project that replicates the crash when using the interrupt pin.  If you can duplicate it, post it in an issue.

**Reply by XanCraft21** [reply]
Ok, i will do that when i can. Thank you. Will it be ok if i post it as a gist?

**Reply by mrengineer7777** [reply]
A gist would be good.  You can create an issue here: https://github.com/espressif/arduino-esp32/issues

**Reply by XanCraft21** [reply]
Thank you. I will begin working on that.

**Reply by XanCraft21** [reply]
Unfortunately i can’t replicate the bug, the entire deep sleep control software is buggy.

i don’t get the interrupt bug, but there are some other bugs i found. If the gpio level to wake up is set to low, the chip instantly wakes up as soon as it goes to sleep, but if i set the level to high it never wakes up even if i keep the pin powered with the correct voltage polarity.

the first but i was having with the watchdog timer uses the normal setup but a different interrupt that is not being triggered.

I tested the sleep on the normal esp32 board, but i tested it first on a different board.

The other board is the Lilygo watch v2 and it seems hard to make software for it since the software that lilygo made sucks badly. And there is a full OS that already exists but it sucks even worse.

**Comment by 27rabbitlt** [comment]
Not sure if we are facing the same problem. But the phenomenon is basically the same.

Everytime when esp board try to execute `esp_deep_sleep_start`, it will be woken up by Interrupt Watchdog, the wakeup cause is 0 (ESP_SLEEP_WAKEUP_UNDEFINED), and the reset reason is 5 (ESP_RST_INT_WDT).

I tried the demo suggested by @mrengineer7777  https://github.com/espressif/arduino-esp32/blob/master/libraries/ESP32/examples/DeepSleep/TimerWakeUp/TimerWakeUp.ino and the demo works perfectly fine.

The only difference is that I missed the `Serial.flush()`. When I flushed the buffer of Serial, and then goto deep sleep, the problem is solved magically.
